### PR TITLE
layered-approach-in-terraform

### PR DESCRIPTION
**Pull request by atiknn**

### CODE REVIEW

File: level1/provider.tf [hunk 1]
```diff
@@ -11,10 +11,10 @@ terraform {
   }
 
   backend "s3" {
-    bucket         = "terraformstatebucket"
-    key            = "terraform.tfstate"
+    bucket         = "terraformstatebucketl1"
+    key            = "level1.tfstate"
     region         = "us-east-1"
-    dynamodb_table = "terraformstate"
+    dynamodb_table = "terraformstatedynamo"
```
Comment (vladimirmukhin) [inline level1/provider.tf:17]: Please let me know why you're changing this?

File: level2/data.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+data "terraform_remote_state" "level1" {
+  backend = "s3"
+
+  config = {
+    bucket = "terraformstatebucketl1"
+    key    = "level1.tfstate"
+    region = "us-east-1"
+  }
+}
```
Comment (vladimirmukhin) [inline level2/data.tf]: Newline is missing

File: level2/terraform.tfvars [hunk 3]
```diff
@@ -0,0 +1 @@
+environment_code = "dev"
```
Comment (vladimirmukhin) [inline level2/terraform.tfvars]: Newline is missing

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: level1/provider.tf (hunk 1 above)
Comment (atiknn) [reply level1/provider.tf:17]: sorry i dont remember, may be because it was giving some error.

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**